datafusion/core/src/physical_plan/joins/cross_join.rs

```diff
@@ -61,34 +61,25 @@ pub struct CrossJoinExec {
 }
 
 impl CrossJoinExec {
-    /// Tries to create a new [CrossJoinExec].
-    /// # Error
-    /// This function errors when left and right schema's can't be combined
-    pub fn try_new(
-        left: Arc<dyn ExecutionPlan>,
-        right: Arc<dyn ExecutionPlan>,
-    ) -> Result<Self> {
-        let left_schema = left.schema();
-        let right_schema = right.schema();
-        check_join_is_valid(&left_schema, &right_schema, &[])?;
-
-        let left_schema = left.schema();
-        let left_fields = left_schema.fields().iter();
-        let right_schema = right.schema();
-
-        let right_fields = right_schema.fields().iter();
-
+    /// Create a new [CrossJoinExec].
+    pub fn new(left: Arc<dyn ExecutionPlan>, right: Arc<dyn ExecutionPlan>) -> Self {
         // left then right
-        let all_columns = left_fields.chain(right_fields).cloned().collect();
+        let all_columns = {
+            let left_schema = left.schema();
+            let right_schema = right.schema();
+            let left_fields = left_schema.fields().iter();
+            let right_fields = right_schema.fields().iter();
+            left_fields.chain(right_fields).cloned().collect()
+        };
 
         let schema = Arc::new(Schema::new(all_columns));
```

Comment:
Could you use `Schema::merge` here instead?

Reply:
You could do something like this. This would preserve schema metadata as well.

```rust
        let input_schemas = vec![left.schema().as_ref().clone(), right.schema().as_ref().clone()];
        let schema = Arc::new(Schema::try_merge(input_schemas)?);
```

Reply:
Sorry, I'm afraid `try_merge` doesn't fit this context.
What we want here is to **concatenate** schema, but not **merge** schema. It could happen that a table cross joins with itself, or two tables have same named columns, in which cases, `merge` doesn't work.

Reply:
For the metadata, we should also do concatenation, instead of merging.

Reply:
@andygrove, what's your suggestion?